rviz_common/src/rviz_common/visualizer_app.cpp

```diff
@@ -147,108 +150,110 @@ bool VisualizerApp::init(int argc, char ** argv)
+      "Enable the Ogre.log file (output in cwd) and console output.");
+  parser.addOption(ogre_log_option);
+
+  // TODO(botteroa-si): enable when possible
+//  QCommandLineOption splash_screen_option(
+//    QStringList() << "s" << "splash-screen",
+//    "A custom splash-screen image to display",
+//    "splash_path");
+//  parser.addOption(splash_screen_option);
+//
+//  QCommandLineOption help_file_option(
+//    "help-file", "A custom html file to show as the help screen", "help_path");
+//  parser.addOption(help_file_option);
+//
+//  QCommandLineOption open_gl_option(
+//    "opengl",
+//    "Force OpenGL version (use '--opengl 210' for OpenGL 2.1 compatibility mode)",
+//    "version");
+//  parser.addOption(open_gl_option);
+//
+//  QCommandLineOption disable_anti_aliasing_option(
+//    "disable-anti-aliasing", "Prevent rviz from trying to use anti-aliasing when rendering.");
+//  parser.addOption(disable_anti_aliasing_option);
+//
+//  QCommandLineOption no_stereo_option("no-stereo", "Disable the use of stereo rendering.");
+//  parser.addOption(no_stereo_option);
+//
+//  QCommandLineOption log_level_debug_option(
+//    "log-level-debug", "Sets the ROS logger level to debug.");
+//  parser.addOption(log_level_debug_option);
+
+//   ("in-mc-wrapper", "Signal that this is running inside a master-chooser wrapper")
+
+  QString display_config, fixed_frame, splash_path, help_path;
+  bool enable_ogre_log;
+  // TODO(botteroa-si): enable when possible
+//  bool in_mc_wrapper = false;
+//  int force_gl_version = 0;
+//  bool disable_anti_aliasing = false;
+//  bool disable_stereo = false;
+
+  try {
+    parser.process(*app_);
+
+    enable_ogre_log = parser.isSet(ogre_log_option);
+//    disable_stereo = parser.isSet(no_stereo_option);
+//    disable_anti_aliasing = parser.isSet(disable_anti_aliasing_option);
+
+    if (parser.isSet(display_config_option)) {
+      display_config = parser.value(display_config_option);
+    }
+    if (parser.isSet(fixed_frame_option)) {
+      fixed_frame = parser.value(fixed_frame_option);
+    }
 
-  rviz_common::install_rviz_rendering_log_handlers();
+    // TODO(botteroa-si): enable when possible
+//    if (parser.isSet(splash_screen_option)) {
+//      splash_path = parser.value(splash_screen_option);
+//    }
+//    if (parser.isSet(help_file_option)) {
+//      help_path = parser.value(help_file_option);
+//    }
+//    if (parser.isSet(splash_screen_option)) {
+//      splash_path = parser.value(splash_screen_option);
+//    }
+//    if (parser.isSet(open_gl_option)) {
+//      force_gl_version = parser.value(open_gl_option).toInt();
+//    }
+
+//   if (vm.count("in-mc-wrapper")) {
+//     in_mc_wrapper = true;
+//   }
+//
+//
+//   if (vm.count("log-level-debug")) {
+//     if (
+//       ros::console::set_logger_level(ROSCONSOLE_DEFAULT_NAME, ros::console::levels::Debug))
+//     {
+//       ros::console::notifyLoggerLevelsChanged();
+//     }
+//   }
+  } catch (std::exception & e) {
```

Comment:
It might be nice to have a more specific try-catch here, maybe just whatever Qt throws if it fails to parse correctly. Could be done in a follow up pr though.

Reply:
It actually turned out that the method `QCommandLineParser::process()` takes care internally of the exception handling, so that no try-catch block is needed. Therefore we just removed it.